src/sharing/SharingScreen.js

```diff
@@ -53,19 +54,21 @@ export default function SharingScreen(props: Props): Node {
   const { params } = props.route;
   const hasAuth = useGlobalSelector(getHasAuth);
 
-  // If there is no active logged-in account, abandon the sharing attempt,
-  // and present the account picker screen to the user.
-  if (!hasAuth) {
-    NavigationService.dispatch(navigateToAccountPicker());
```

Comment:
Keeping a SharingScreen around and rendering, while showing the
    account picker, could plausibly have contributed to a "No server
    data found" error we've seen in Sentry in recent releases:
      https://sentry.io/organizations/zulip/issues/2787518778/?project=191284
    In particular, it's possible to remove server data by interacting
    with the account picker: you can tap the trash can to make an
    account disappear. See the previous commit for another factor that I
    think would have to be active to cause this error; we fixed that
    factor there.

This change seems good for the other reason you mention, but I don't think it can be related to that Sentry issue. (I think the first commit probably does fix the issue.)

In that issue, the stack shows we're still in the middle of navigating *to* the sharing screen:
```
  at goToSharing(src/sharing/index.js:15:21)
```
That's this line:
```js
  NavigationService.dispatch(navigateToSharing(data));
```

So I think it's actually about the first point mentioned in the first commit:

    - The share-to tabs need more than the guarantee of a logged-in
      user. They depend on server data like `realm.mandatoryTopics`.